### PR TITLE
Footstep audio

### PR DESCRIPTION
**Pull request by duriumm**
- Footstep sound added for grass and dirt. To add more sounds just import them into Resource folder and then reference them in the script
- This works by using autotiling and setting the tiles Tag to "Tile". Then depending on the tiles name we play a sound 

image

### CODE REVIEW

File: Pixel Game/Assets/FootstepAudioSwitcher.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    // Here we run the audio switching function when we encounter a new tile collider.
+    // Get the tiles name (dirt, grass, stone) to input as value to audioManager
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if(collision.gameObject.tag == "Tile")
+        {
+            audioManager.SwitchSoundType(collision.name);
+        }
```
Comment (yousernaym) [inline Pixel Game/Assets/FootstepAudioSwitcher.cs:31]: Very nice solution

File: Pixel Game/Assets/Scripts/AudioManager.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public void PlaySound()
+    {
+        audioSource.pitch = Random.Range(0.90f, 1.10f);
+        audioSource.PlayOneShot(currentActiveFootstepSound);
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/AudioManager.cs:34]: Nice touch with the varying pitch

File: Pixel Game/Assets/Scripts/AudioManager.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public void TurnOnThenOffAudioFootstepCollider()
+    {
+        footStepColliderObject.SetActive(false);
+        footStepColliderObject.SetActive(true);
+    }
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/AudioManager.cs]: The name seems backwards. It's turning off then on.

File: Pixel Game/Assets/Scripts/AudioManager.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    // Not great using magic strings here so preferably an enum for future use in audioManager
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/AudioManager.cs:42]: 👍

File: Pixel Game/Assets/Scripts/PlayerScripts/PlayerMovement.cs [hunk 5]
```diff
@@ -1,10 +1,15 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Tilemaps;
 
 public class PlayerMovement : Movement
 {
+    // Tried making a overridden Start() function but did not manage, therefor 
+    // i use a public audioManager dragged in from the hirearchy
+    public AudioManager audioManager;
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/PlayerScripts/PlayerMovement.cs:10]: As long as it works. For some reason the base Start is protected, so the overriden Start needs to be protected too. Just guessing that was the problem :)

File: Pixel Game/Assets/Scripts/AudioManager.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    public void SwitchSoundType(string typeOfGround)
+    {
+        switch (typeOfGround)
+        {
+            case "Dirt":
+                Debug.Log("it was dirt");
+                currentActiveFootstepSound = dirt_footstep;
+                break;
+            case "Grass":
+                Debug.Log("it was grass in my ass");
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/AudioManager.cs]: Probably a good idea to remove or comment out the debug logs

**Review by yousernaym — COMMENTED** [review]
Woohoo, another PR!!!

File: Pixel Game/Assets/Scripts/AudioManager.cs (hunk 3 above)
Comment (duriumm) [reply Pixel Game/Assets/Scripts/AudioManager.cs]: Fixed !

File: Pixel Game/Assets/Scripts/AudioManager.cs (hunk 6 above)
Comment (duriumm) [reply Pixel Game/Assets/Scripts/AudioManager.cs]: Fixed!

**Pull request merged**